tests/testthat/test-sql-escape.r

```diff
@@ -42,3 +42,27 @@ test_that("times are converted to ISO 8601", {
   x <- ISOdatetime(2000, 1, 2, 3, 4, 5, tz = "US/Central")
   expect_equal(escape(x), sql("'2000-01-02T09:04:05Z'"))
 })
+
+# Name aliasing -----------------------------------------------------------
+
+test_that("queries generated by select() don't alias unnecessarily", {
+  lf_build <- lazy_frame(x = 1) %>%
+    select(x) %>%
+    sql_build()
+
+  render_odbc <- sql_render(lf_build, con = simulate_odbc())
+  render_dbi  <- sql_render(lf_build, con = simulate_dbi())
+
+  expect_equal(render_odbc, sql("SELECT `x`\nFROM `df`"))
+  expect_equal(render_dbi,  sql("SELECT \"x\"\nFROM \"df\""))
+})
+
+test_that("names_to_as() doesn't alias when ident name and value are identical", {
```

Comment:
Could you add a similar test for when the names are missing?  And one to make sure that the names are used, when present? That's what I meant by testing the existing behaviour as well.